### PR TITLE
Fixes for build on FreeBSD

### PR DESCRIPTION
**Pull request by ppekala**
- Add OD_MAN_PATH variable allowing customization of manpages install prefix
- Link with libexecinfo needed for backtrace functions
- Patch StackTraceUnix.cpp similar to OpenBSD

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -32,6 +32,7 @@ else()
     # Set binary and data install locations if we want to use the installer
     set(OD_BIN_PATH ${CMAKE_INSTALL_PREFIX}/games CACHE PATH "Absolute path to the game binary directory")
     set(OD_DATA_PATH ${CMAKE_INSTALL_PREFIX}/share/games/${PROJECT_NAME} CACHE PATH "Absolute path to the game data directory")
+    set(OD_MAN_PATH ${CMAKE_INSTALL_PREFIX}/share CACHE PATH "Absolute path to the manpages directory")
```
Comment (akien-mga) [inline CMakeLists.txt]: The default value should be `${OD_SHARE_PATH}/man`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -776,7 +782,7 @@ if(UNIX)
     install(FILES ${CMAKE_BINARY_DIR}/opendungeons.desktop
             DESTINATION ${OD_SHARE_PATH}/applications)
     install(FILES ${CMAKE_BINARY_DIR}/opendungeons.6
-            DESTINATION ${OD_SHARE_PATH}/man/man6)
+	    DESTINATION ${OD_MAN_PATH}/man/man6)
```
Comment (akien-mga) [inline CMakeLists.txt]: Indentation issue here.

File: CMakeLists.txt [hunk 3]
```diff
@@ -629,6 +630,11 @@ target_link_libraries(
     ${EXTRA_LIBRARIES}
 )
 
+# backtrace functions are not part of libc like on Linux platform
+if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
```
Comment (akien-mga) [inline CMakeLists.txt]: Do you know if it would be needed on OpenBSD too?

Comment (ppekala) [reply CMakeLists.txt]: OpenBSD does not have execinfo implementation in libc too, so yes. But FreeBSD have libexecinfo included in base system while OpenBSD need to install it as a port. This means
OpenBSD would need check for libexecinfo, and probably setting includes for build (execinfo.h in /usr/include vs in /usr/local/include). I'm starting to speculating here and don't want add to the patch anything I'm not sure 100%.

**Comment by ppekala** [comment]
Didn't noticed that #1282 already added libexecinfo detection, so I removed last commented part.

**Comment by tomluchowski** [comment]
(just greetings in polish :D ) : 
@ppekala  Cześć witam rodaka :) , może Panu się spodoba nasz projekt i będzie Pan nie tylko dodawał ulepszenia dla FreeBSD , ale też innych aspektów gry :) pozdrawiam Tomasz Luchowski

**Comment by ppekala** [comment]
@tomluchowski Być może :) Dużo projektów mało czasu... Pozdrawiam

**Comment by akien-mga** [comment]
Sorry for the late (re-)review, it looks fine. I've squashed the commits manually into 35b1c131.

**Pull request closed without merging**